### PR TITLE
[ENH] Beta Distribution

### LINKED ISSUE #22: [ENH] (wish)list of probability distributions to implement
It would be great to have a basic set of probability distributions implemented.

Umbrella issue for implementing `sktime` probability distributions.

High priority:

- [ ] laplace
- [ ] empirical incl delta
- [ ] multivariate normal

mid priority:

- [ ] t-distribution
- [ ] Cauchy (special case of Cauchy)
- [ ] mixture composition
- [ ] truncated, compositor

low priority:

- [ ] uniform continuous
- [ ] beta
- [ ] chi-squared
- [ ] logistic
- [ ] log-normal
- [ ] Weibull

Should extend `BaseDistribution`, a good template is `Normal`.

Mirrors https://github.com/sktime/sktime/issues/4518
(for high and mid priority)

### PR DESCRIPTION
**Pull request by malikrafsan**
#### Reference Issues/PRs

Towards #22

#### What does this implement/fix? Explain your changes.

This PR implements Beta distribution based on Scipy Adapter

#### What should a reviewer concentrate their feedback on?

- [x] Whether the Scipy Adapter makes it easier to implement new distribution

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/skpro/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/skpro/blob/main/.all-contributorsrc) in the `skpro` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [x] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [x] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

File: docs/source/api_reference/distributions.rst [hunk 1]
```diff
@@ -42,6 +42,7 @@ Continuous support
     Normal
     TDistribution
     Weibull
+    Beta
```
Comment (fkiraly) [inline docs/source/api_reference/distributions.rst:45]: let's order these by alphabet

File: skpro/distributions/__init__.py [hunk 2]
```diff
@@ -19,8 +19,10 @@
     "TDistribution",
     "Uniform",
     "Weibull",
+    "Beta",
```
Comment (fkiraly) [inline skpro/distributions/__init__.py:22]: let's order these by alphabet

File: skpro/distributions/beta.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+import pandas as pd
+from scipy.stats import beta, rv_continuous
+
+from skpro.distributions.adapters.scipy import _ScipyAdapter
+
+
+class Beta(_ScipyAdapter):
+    r"""Beta distribution.
+
```
Comment (fkiraly) [inline skpro/distributions/beta.py:14]: could you add a short note about parameterization? Have a look at the other distributions.

The reason is, many distributions have different parameterizations, e.g., normal can be parameterized by sigma or sigma-squared. It suffices to spell out one of the distribution defining functions, usually I choose the "simplest" for users to understand.

Let me know if you are unsure, then I will add it later.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Excellent - it looks like we have minimized the effort to add more distributions with the adapter!

I suppose this takes 5 or 10 min per distribution now?

I left a few comments.

**Comment by fkiraly** [comment]
> Whether the Scipy Adapter makes it easier to implement new distribution

Well, you tell me - you implemented it, was it easier?

File: docs/source/api_reference/distributions.rst (hunk 1 above)
Comment (malikrafsan) [reply docs/source/api_reference/distributions.rst]: Fixed, thank you for pointing this out!

File: skpro/distributions/__init__.py (hunk 2 above)
Comment (malikrafsan) [reply skpro/distributions/__init__.py]: Fixed, thank you for pointing this out!

File: skpro/distributions/beta.py (hunk 3 above)
Comment (malikrafsan) [reply skpro/distributions/beta.py:14]: Recently, I have added the parameterization note about the beta distribution. I had a look at the other distributions, and a lot of them are noted by PDF or CDF formula. That's why I tried to add the PDF formula as I think it is easier to understand, in the context of Beta Distribution. I took the formula from the [Wikipedia](https://en.wikipedia.org/wiki/Beta_distribution). I added the PDF formula as given

$$f(x) = \frac{x^{\alpha-1}(1-x)^{\beta-1}}{B(\alpha, \beta)}$$

Please let me know if there is something I missed! Also, I am totally fine as well if you want to modify the PR as well

**Comment by malikrafsan** [comment]
> > Whether the Scipy Adapter makes it easier to implement new distribution
> 
> Well, you tell me - you implemented it, was it easier?

Sure! It was much easier compared to when I tried to implement the Weibull / LogLogistic distributions 😅

File: skpro/distributions/beta.py (hunk 3 above)
Comment (fkiraly) [reply skpro/distributions/beta.py:14]: yes, that works, especially with the explanation that B is for normalization.

**Review by fkiraly — APPROVED** [review]
All addressed!

**Pull request merged**